utility/src/main/java/org/oppia/util/system/OppiaClock.kt

```diff
@@ -0,0 +1,32 @@
+package org.oppia.util.system
+
+import android.os.SystemClock
+import androidx.annotation.VisibleForTesting
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Utility to get the current date/time. */
+@Singleton
+class OppiaClock @Inject constructor() {
+  private var testTimeMs: Long? = null
+
+  fun getElapsedRealtimeMs(): Long {
+    return SystemClock.elapsedRealtime()
+  }
+
+  fun getCurrentTimeMs(): Long {
+    return testTimeMs ?: System.currentTimeMillis()
+  }
+
+  @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+  fun setCurrentTimeMs(currentTimeMs: Long) {
+    testTimeMs = currentTimeMs
+  }
+
+  fun getCurrentCalendar(): Calendar {
+    val calendar = Calendar.getInstance()
+    calendar.timeInMillis = getCurrentTimeMs()
+    return calendar
```

Comment:
You may directly return `Calendar.getInstance().timeInMillis=getCurrentTimeMs()`

Reply:
we cannot do assignment in return..